enhancements/monitoring/multi-tenant-alerting.md

```diff
@@ -0,0 +1,601 @@
+based on platform and/or application metrics. They are evaluated by the Thanos
+ruler instances (by default) or the Prometheus instances running in the
+`openshift-user-workload` namespace.
+
+When a user alert fires, the Thanos Ruler (or UWM Prometheus) sends it to the
+Plaform Alertmanager cluster (deployed in the `openshift-monitoring` namespace)
+where it gets aggregated and dispatched to the correct destination (page, chat,
+email, ticket, ...).
+
+The configuration of Alertmanager is done via a single configuration file that
+only cluster admins have permissions to modify. If the cluster is shared by
+multiple tenants and each tenant has different requirements to receive their
+notifications then each tenant needs to ask and wait for the cluster admins to
+adjust the Alertmanager configuration.
+
+To streamline the process and avoid cluster admins being the bottleneck,
+application owners should be able to configure alert routing and notification
+receivers in the Plaform Alertmanager without cluster admin intervention.
+
+[AlertmanagerConfig][alertmanagerconfig-crd] CRD fullfills this requirement and
+is supported by the [Prometheus operator][prometheus-operator] since v0.43.0
+but it is explicitly called out in the OCP documentation as ["not
+supported"][unsupported-resources].
+
+### Goals
+
+* Cluster users can configure alert notifications for applications being
+monitored by the user-workload monitoring without requesting intervention from
+cluster admins.
+* Cluster admins can grant permissions to users and groups to manage alert
+routing scoped to individual namespaces.
+* Namespace owners should be able to opt-out from Alertmanager
+configuration (similar to what exist for service/pod monitors and rules using the
+`"openshift.io/user-monitoring: false"` label on the namespace).
+* Cluster admins should be able to opt-out from supporting `AlertmanagerConfig`
+resources from user namespaces.
+
+### Non-Goals
+
+* Additional support for silencing user alerts (it is already supported by UWM in the OCP console).
+* Specific integration in the OCP Console exposing the configuration of alert notifications.
+* Support the configuration of alert notifications for platform alerts (e.g.
+alerts originating from namespaces with the `openshift.io/cluster-monitoring: "true"`
+label).
+* Share alert receivers and routes across tenants.
+* Deploy an additional UWM Alertmanager.
+
+## Proposal
+
+We plan to leverage the `AlertmanagerConfig` custom resource definition already
+exposed by the Prometheus operator so that application owners can configure how
+and where their alert notifications should be routed.
+
+### User Stories
+
+Personas:
+* Application owners: manage a project with sufficient permissions to define monitoring resources.
+* UWM admins: manage the configuration of the UWM components (edit permissions on the `openshift-user-workload-monitoring/user-workload-monitoring-config` configmap).
```

Comment:
Thanks for the info, I wasn't aware of this issue. I don't see an easy solution either...

Reply:
Are you talking about prometheus-operator only? Or UWM Prometheus and Thanos Ruler?
Aren't the control plane's nodes tainted to avoid such situation?

Reply:
It has tolerations by default :) https://github.com/openshift/managed-cluster-config/pull/990

Reply:
Thanks for the details, I understand better now :)